collector/test/ProcessSignalFormatterTest.cpp

```diff
@@ -12,14 +12,34 @@ namespace collector {
 
 using ProcessSignal = ProcessSignalFormatter::ProcessSignal;
 using LineageInfo = ProcessSignalFormatter::LineageInfo;
+using namespace testing;
+
+class MockCollectorConfig : public CollectorConfig {
+ public:
+  MockCollectorConfig() = default;
+
+  void SetDisableProcessArguments(bool value) {
+    disable_process_arguments_ = value;
+  }
+};
+
+class MockProcessSignalFormatter : public ProcessSignalFormatter {
+ public:
+  MockProcessSignalFormatter(sinsp* inspector, const CollectorConfig& config) : ProcessSignalFormatter(inspector, config) {};
+
+  ProcessSignal* MockCreateProcessSignal(sinsp_evt* event) {
+    return CreateProcessSignal(event);
+  }
+};
```

Comment:
Could we use `FRIEND_TEST` definitions here instead of mocks? Ideally, we would have an easy way to create a config object with any configuration we want without any of these methods, but for now, I think I prefer not having mocks when all we are using them for is access to private/protected fields, WDYT?

http://google.github.io/googletest/reference/testing.html#FRIEND_TEST